### PR TITLE
Create text component

### PR DESCRIPTION
**Pull request by amberleyromo**
# Description

Create `<Text>` component in gatsby-interface to replace shared components in dotcom `Typography` file.

# Notes

- `VARIANTS`
	- `PRIMARY` (default): Empty object, doesn't build on the base styles. Left it in for completeness.
	- `EMPHASIZED`: Bolded font.
	- `ERROR`: Pulls over from mansion. Quietly expects an icon, but doesn't break without one.
- `TONES`
	- `NEUTRAL` (default)
	- `BRAND`
- `SIZES`
	- `S`
	- `M` (default)
	- `L`
	- `XL`
- `BaseText` component allows `as` `p` or `span`.
- Margins: Not setting margins in component code, allowing for native margins.

# Screenshots

## `as`

Screen Shot 2019-11-08 at 10 03 49 AM

## variants

Screen Shot 2019-11-08 at 10 03 58 AM

## tones

Screen Shot 2019-11-08 at 10 04 06 AM

## sizes

Screen Shot 2019-11-08 at 10 04 20 AM

## with children

Screen Shot 2019-11-08 at 10 05 00 AM

# Discussion

- [x] For `as` prop, I allowed `p` and `span`. I don't really think we should allow `div` tags (@greglobinski? https://github.com/gatsby-inc/gatsby-interface/issues/86#issuecomment-546419364).
- [x] Curious what y'all think about the sizing. Using the approach we're using in `core/Button`. **What scale should we use for the line height changes**?

# Related

Issue #86

### CODE REVIEW

File: src/theme/styles/text.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      },
+    },
+  }
+}
+
+const sizeStyles = () => {
+  return {
+    S: {
+      fontSize: fontSizes[1],
+      lineHeight: 1.5,
```
Comment (amberleyromo) [inline src/theme/styles/text.js]: What scale should we use for lineHeight?

Comment (amberleyromo) [reply src/theme/styles/text.js]: mob programming: it should remain constant.

**Comment by amberleyromo** [comment]
Mob programming:
- line-height should remain constant (not scale)
- We should include a `lede` variant in this component, rather than creating a new, separate component.

**Comment by rizchelwood** [comment]
Is this complete/ready for review? If it is, could you add reviewers so we know it's ready

**Comment by amberleyromo** [comment]
@rizchelwood this has been sitting waiting for review and i didn't see your comment 🤦🏻‍♀️. Super non-urgent though.

File: src/theme/styles/text.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    "2XL": {
+      fontSize: fontSizes[5],
+      lineHeight: 1.5,
+    },
+  }
+}
+
+const variantStyles = () => {
+  console.log(`sizeStyles()[M]`, sizeStyles()[`M`])
```
Comment (Elanhant) [inline src/theme/styles/text.js]: Some console logs crawled in 🙂

File: src/theme/styles/text.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      margin: `0.5rem 0 0`,
+    },
+    EMPHASIZED_LEDE: {
+      fontFamily: fonts.header.join(`,`),
+      ...sizeStyles()[`2XL`],
+      lineHeight: 1.3,
+      margin: `1em 0 0`,
+    },
+    ERROR: {
+      svg: {
```
Comment (Elanhant) [inline src/theme/styles/text.js]: Let's not target `svg` here. If we ever want to display an icon in an ERROR text, it's better to create a custom component for that.

File: src/theme/styles/text.js (hunk 2 above)
Comment (amberleyromo) [reply src/theme/styles/text.js]: Oh drat.

**Comment by amberleyromo** [comment]
@Elanhant re: the failing check, should this still be pointing at `dev`? Confirming how your recent changes to the flow should affect this PR.

**Comment by Elanhant** [comment]
> @Elanhant re: the failing check, should this still be pointing at `dev`? Confirming how your recent changes to the flow should affect this PR.

The check fails because this branch was created before we've got a CircleCI integration. Could you please merge latest `dev`? This should fix the check failing. 

Also, for now let's keep it to `dev`, I'll let everyone know once we switch to `master` as primarily merge destination :)

**Pull request merged**